src/wallet/rpcwallet.cpp

```diff
@@ -3633,8 +3652,8 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
         txsize += CTXOUT_REGULAR_SIZE;      // There will probably be taddr change
     }
     txsize += CTXOUT_REGULAR_SIZE * taddrRecipients.size();
-    if (txsize > MAX_TX_SIZE) {
-        throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Too many outputs, size of raw transaction would be larger than limit of %d bytes", MAX_TX_SIZE ));
+    if (txsize > max_tx_size) {
```

Comment:
Seems, the computation of `max_tx_size` in all three methods should be refactored somewhere, especially since it gets a little messy with looking at what are the active upgrades, which also distracts from what else is going on in the method.
Btw, I'm wondering why the transparent sending methods like `sendrawtransaction` do not require similar changes.

Reply:
(I'm aware on argument for not refactoring is that some of this is temporary till upgrade after Sapling activation; still, I'm worried about the code getting entangled with these if's relating to what are the active upgrades)